### DISCUSSION TITLE
Pd64 icon for easier recognizability

### DISCUSSION THREAD

**Discussion by ben-wes** [pure-data/pure-data#2330] [Ideas]
While there's a bit of talk around Pd64 lately (on the list and https://github.com/pure-data/pure-data/issues/2329), i'm adding this here ...

I've been working with Pd64 on macOS arm64 (Macbook Air M2, Sonoma 14.x) since December 2023 now (as a side note: i appreciate it a lot and it worked very reliable for me so far!). Since a few externals are not yet available there (mainly thinking of `Gem` and `ELSE` in my case), i did not completely switch yet and i assume, other users/testers also currently have parallel Pd/Pd64 versions installed.

While using both versions, it quickly started bothering me that i couldn't visually distinguish between the two in the taskbar or when switching apps - so i tried to solve this for me by just exchanging icons in a branch for Pd64 and compiling locally. (i also changed `plist` management to allow separate external directories - but later learnt that this is not necessary.)

The icon changes (and hacky plist adaptions for macOS) are here:
https://github.com/pure-data/pure-data/compare/master...ben-wes:pure-data:pd64-preferences

... now i don't think a new color is really necessary and the dirty white was actually by mistake. But maybe an additional `64` might be helpful also for the official build?

I'm attaching the (zipped) `icns` file as a suggestion here:
[pd64.icns.zip](https://github.com/user-attachments/files/15523115/pd64.icns.zip)

Preview:
<img width="64" alt="icon_128x128" src="https://github.com/pure-data/pure-data/assets/1923787/b766a1bb-5c53-4ff8-a2d2-1749d3c65fed">

**Comment by porres** [comment]
nice!

**Comment by umlaeute** [comment]
hmm. i think the simplest way to distributing Pd64 is adding it to the application bundles.
we already do this for the flatpak packages, and we could switch the macOS application bundles at any time.
it's also rather simple for the Windows installers/zips, but afaik miller builds them himself (rather than taking them from the CI), so might require a bit more persuasion.

in any case: if the application comes with both Pd64 and Pd32, i don't think there is a need for a separate icon.

**Reply by ben-wes** [reply]
> if the application comes with both Pd64 and Pd32, i don't think there is a need for a separate icon.

... isn't the simultaneous distribution of 2 almost identical applications exactly the case that creates the need for a clear distinction between them? - since it's quite possible that they will be in the same taskbar, on the same "desktop" and running at the same time? (they obviously have different names, but the visual hint might be parsed faster)

to clarify: i didn't intend this as an icon for app distribution. that might be a separate discussion after having a look at https://flathub.org/apps/search?q=puredata (i have no idea how to avoid confusion in such cases though, since i don't question Pd's icon).

i'm not sure i understand what the result for users will be when installing a Pd/Pd64 bundle on Windows or macOS - so my assumptions here might be wrong if this doesn't automatically mean that they have both versions installed in parallel.

**Reply by umlaeute** [reply]
> ... isn't the simultaneous distribution of 2 almost identical applications

the idea is to not distribute 2 almost identical applications, but a single application where you can select the floatsize via the preferences.

- the flatpak already does this.
- we can switch the macOS app to do that (after the 0.55 release)
- we can also build the Windows installer/zip to do that (but iirc, the Windows releases are built by Miller, so we have to pursuade him)
- heck. i think i already wrote this.

> i'm not sure i understand what the result for users will be when installing a Pd/Pd64 bundle on Windows or macOS - so my assumptions here might be wrong if this doesn't automatically mean that they have both versions installed in parallel.

the result is, that you can chose the floatsize in the preferences:

![startup preferences](https://hosted.weblate.org/media/screenshots/preferences-startup.png)

**Reply by ben-wes** [reply]
> [...] but a single application where you can select the floatsize via the preferences.

oha, i didn't consider that! that certainly renders the concept of different icons quite useless.

> * the flatpak already does this.

damn, i guess, i'm on the wrong system.

> * we can switch the macOS app to do that (after the 0.55 release)

ok, there's still hope for my system.

> * heck. i think i already wrote this.

sounded familiar, yes. :) but took a bit for me to understand what you meant - sorry.

(will close the discussion and keep the icons for my own use until this is implemented)
